packages/tldraw/src/lib/shapes/shared/useAsset.ts

```diff
@@ -1,38 +1,104 @@
-import { TLAssetId, TLShapeId, useEditor, useValue } from '@tldraw/editor'
-import { useEffect, useState } from 'react'
+import {
+	MediaHelpers,
+	TLAssetId,
+	TLImageShape,
+	TLShapeId,
+	TLVideoShape,
+	useEditor,
+	useReactor,
+} from '@tldraw/editor'
+import { useRef, useState } from 'react'
 
-/** @internal */
-export function useAsset(shapeId: TLShapeId, assetId: TLAssetId | null, width: number) {
+function useScalableAsset(shapeId: TLShapeId, assetId: TLAssetId | null) {
 	const editor = useEditor()
-	const [url, setUrl] = useState<string | null>(null)
 	const asset = assetId ? editor.getAsset(assetId) : null
-	const culledShapes = editor.getCulledShapes()
-	const isCulled = culledShapes.has(shapeId)
-
-	const shapeScale = asset && 'w' in asset.props ? width / asset.props.w : 1
-	// We debounce the zoom level to reduce the number of times we fetch a new image and,
-	// more importantly, to not cause zooming in and out to feel janky.
-	const screenScale = useValue('zoom level', () => editor.getZoomLevel() * shapeScale, [
-		editor,
-		shapeScale,
-	])
-
-	useEffect(() => {
-		if (isCulled) return
-
-		let isCancelled = false
-		const timer = editor.timers.setTimeout(async () => {
-			const resolvedUrl = await editor.resolveAssetUrl(assetId, {
-				screenScale,
-			})
-			if (!isCancelled) setUrl(resolvedUrl)
-		}, 500)
-
-		return () => {
-			clearTimeout(timer)
-			isCancelled = true
-		}
-	}, [assetId, asset?.props.src, isCulled, screenScale, editor])
-
-	return { asset, url }
+
+	// The resolved url of the asset that should be used by the shape (duplicated as a ref)
+	const [state, setState] = useState<{ url: string | null; isAnimated: boolean }>({
```

Comment:
good idea on just putting `isAnimated` inside here (although it _might_ be a little image specific / unrelated to videos).

i don't think it needs to be part of `state`. it's only computed once and then just copy-pasted from then on. let's keep it simple and just have a state for the url and just return the `isAnimated`.

i.e.
```
const [url, setUrl] = useState(null)
const [isAnimated, setIsAnimated] = useState(() =>
  ('mimeType' in asset.props && MediaHelpers.isAnimatedImageType(asset?.props.mimeType)) ||
  ('isAnimated' in asset.props && asset.props.isAnimated))
```